### ISSUE TITLE
Support IDM login

### ISSUE THREAD

**Issue by froqstar** [element-hq/element-x-android#3789]
#### What would you like to do?

I'd like to login using IDM instead of plain username/password

#### Why would you like to do it?

My identity server only supports IDM login now, which prohibits me from using Element X, as I can't login.

#### How would you like to achieve it?

Element (without X) supports this, I don't know whether any code can be reused from there.

### Have you considered any alternatives?

Login via QR code from desktop would be a workaround, but it says "Your account provider doesn't support signing into a new device with a QR code".

### Are you willing to provide a PR?

No

**Comment by Somaxax** [comment]
Same problem, no way to login with a google account

**Comment by gepbird** [comment]
Same problem, I can't use Element X at all with my current account

**Comment by ShadowRZ** [comment]
@froqstar @Somaxax @gepbird In Element X, IDM login only supports being backed by [MAS](https://github.com/element-hq/matrix-authentication-service), and it will never support Synapse builtin SSO which I guess your homeservers only supports.

If you're on matrix.org it should already support IDM login.

(BTW based on comments I interpreted that what you're really saying is SSO Login, correct me if I'm wrong)

**Comment by gepbird** [comment]
> [@froqstar](https://github.com/froqstar) [@Somaxax](https://github.com/Somaxax) [@gepbird](https://github.com/gepbird) In Element X, IDM login only supports being backed by [MAS](https://github.com/element-hq/matrix-authentication-service), and it will never support Synapse builtin SSO which I guess your homeservers only supports.
> 
> If you're on matrix.org it should already support IDM login.
> 
> (BTW based on comments I interpreted that what you're really saying is SSO Login, correct me if I'm wrong)

Yes, I meant SSO logins like logging in with a GitHub or Google account on matrix.org homeserver, I wasn't sure what is IDM before this, sorry!

https://github.com/element-hq/element-x-android/issues/4138 would be more fitting for our issue, but I'm not sure *why* was it closed, I can't find a similar one or one with MAS

**Comment by ShadowRZ** [comment]
@gepbird Only next gen auth backed by the MAS I linked earlier would be supported by Element X, traditional SSO that you can enable, for example in Synapse according to [Synapse manual](https://element-hq.github.io/synapse/latest/usage/configuration/user_authentication/single_sign_on/index.html) will never be supported, that's why it's being closed. (Referenced issue was talking about matrix.org which only migrated to MAS recently so people can do direct social login in Element X)

**Comment by MrRinkana** [comment]
@ShadowRZ could you clarify what you mean that traditional SSO will never be supported? To my knowledge MAS is only a transition tech like sliding-sync-proxy was? Isn't MSC3861 fully supported in synapse now, which means external Oath2.0 providers are supported, not only MAS?

**Comment by G-Pereira** [comment]
@shadowRZ If it indeed exists can you please provide some link or procedure to enable SSO login (like Authentik) on Element X?
Our server is lcdporto.org and you can check it advertises SSO login with Authentik and Element X does not offer to login with it. I also temporarily tried to disable password auth on synapse and Element X complained it has no suitable method to login.

**Comment by G-Pereira** [comment]
Found it. I didn't undertand it was something to run separately: https://element-hq.github.io/matrix-authentication-service/setup/installation.html
